src/UserSpaceInstrumentation/FindFunctionAddress.cpp

```diff
@@ -35,16 +38,16 @@ ErrorMessageOr<uint64_t> FindFunctionAddress(pid_t pid, std::string_view module_
   }
 
   OUTCOME_TRY(auto&& elf_file, orbit_object_utils::CreateElfFile(module_file_path));
-  auto symbols = elf_file->LoadSymbolsFromDynsym();
+  ErrorMessageOr<DebugSymbols> symbols = elf_file->LoadSymbolsFromDynsym();
   if (symbols.has_error()) {
     return ErrorMessage(absl::StrFormat("Failed to load symbols for module \"%s\": %s",
                                         module_soname, symbols.error().message()));
   }
 
-  for (const orbit_grpc_protos::SymbolInfo& symbol : symbols.value().symbol_infos()) {
-    if (symbol.name() == function_name) {
+  for (const FunctionSymbol& symbol : symbols.value().function_symbols) {
+    if (symbol.demangled_name == function_name) {
```

Comment:
This looks suspicious. Before we were looking for the address of a function based on the mangled name. Now it is the demangled one. While this seem much more consistent, I would like to double check if this is correct.

@danielfenner or @dpallotti, maybe you know better.

Reply:
Yeah, I believe this has no effect as we are only looking for C names (dlopen, dlsym, dlclose; see `InjectLibraryInTracee.cpp`), but indeed I'm wondering why the change. I think this is more for @danielfenner anyway.

Reply:
> Yeah, I believe this has no effect as we are only looking for C names (dlopen, dlsym, dlclose; see `InjectLibraryInTracee.cpp`), but indeed I'm wondering why the change. I think this is more for @danielfenner to see what his original intent is on mangled vs unmangled, if any.

Reply:
If this is the case, that would be great. We probably might be able to remove mangled names completely (I think this was the only real occurrence left).